python/src/equistore/operations/ones_like.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Union
+
+from equistore import TensorBlock, TensorMap
+
+from . import _dispatch
+from ._utils import _check_parameters_in_gradient_block
+
+
+def ones_like(
+    tensor: TensorMap,
+    parameters: Union[List[str], str] = None,
+    requires_grad: bool = False,
+) -> TensorMap:
+    """Create a ones_like :py:class:`TensorMap` with the same metadata of the input
+    :py:class:`TensorMap` (tensor), but all the values equal to zero.
+
+    All the blocks.values are full of ones and the metadata are given as input.
+
+    :param tensor: input tensor from which you get the metadata.
+    :param parameters: which gradient parameter to copy. Default ``None``
+                      means copy all of them if present.
+                      Empty list for no gradients information.
+    :param requires_grad: If autograd should record operations on the returned tensor.
+                          Only for torch, default False.
+    """
+
+    blocks = []
+    for _k, block in tensor:
```

Comment:
Since you don't need k can you not loop over the blocks directly?

```suggestion
    for block in tensor.blocks():
```